src/components/organize/TagsManager/TagSelect.tsx

```diff
@@ -1,51 +1,147 @@
-      }
-      getOptionLabel={(option) => option.title}
-      groupBy={(option) =>
-        option.group?.title ||
-        intl.formatMessage({
-          id: 'misc.tags.tagsManager.ungroupedHeader',
-        })
-      }
-      onChange={(e, value) => {
-        if (value) {
-          onSelect(value);
-        }
-      }}
-      openOnFocus
-      options={tags}
-      renderInput={(params) => (
-        <TextField
-          {...params}
-          data-testid="TagsManager-tagSelectTextField"
-          placeholder={intl.formatMessage({
-            id: 'misc.tags.tagsManager.addTag',
-          })}
-          variant="outlined"
-        />
-      )}
-      renderOption={(option) => {
-        return <TagChip tag={option} />;
-      }}
-      style={{ width: 300 }}
-    />
+    <Box {...getRootProps()}>
+      <TextField
+        {...getInputProps()}
+        autoFocus
+        fullWidth
+        inputProps={{
+          'data-testid': 'TagManager-TagSelect-searchField',
+        }}
+        placeholder={intl.formatMessage({
+          id: 'misc.tags.tagsManager.addTag',
+        })}
+        variant="outlined"
+      />
+      {/* Options */}
+      <List
+        {...getListboxProps()}
+        style={{ maxHeight: '400px', overflowY: 'scroll' }}
+      >
+        {Object.values(groupedFilteredTags).map((group) => {
+          // Groups
+          return (
+            <List
+              key={group.title}
+              subheader={
+                <ListSubheader component="div" disableSticky>
+                  {group.title}
+                </ListSubheader>
+              }
+              title={group.title}
+            >
+              {/* Tags */}
+              {group.tags.map((tag) => {
+                return (
+                  <ListItem
+                    key={tag.id}
+                    button
+                    disabled={disabledTags
+                      .map((disabledTags) => disabledTags.id)
+                      .includes(tag.id)}
+                    onClick={() => onSelect(tag)}
+                    tabIndex={-1}
+                  >
+                    <TagChip
+                      chipProps={{ style: { cursor: 'pointer' } }}
+                      tag={tag}
+                    />
+                  </ListItem>
+                );
+              })}
+            </List>
+          );
+        })}
+        <ListItem
+          button
+          data-testid="TagManager-TagSelect-createTagOption"
+          onClick={() =>
+            setDialogOpen({
+              defaultTitle: inputValue,
+              open: true,
+            })
+          }
+        >
+          <Add />
+          {inputValue ? (
+            <FormattedMessage
+              id="misc.tags.tagsManager.createNamedTag"
+              values={{
+                b: (...chunks: string[]) => (
+                  <>
+                    &nbsp;<b>{chunks}</b>
+                  </>
+                ),
```

Comment:
I did not know it was possible to do this!

Reply:
It's not the prettiest way but for some reason the intl library wasn't rendering the space before the bolded text, so it's a bit of an ugly workaround but it does the job... :( Not really proud of that one.

Reply:
i've done this too haha not beautiful but get's the job done!